src/hypertable.c

```diff
@@ -2402,6 +2402,67 @@ ts_hypertable_get_open_dim_max_value(const Hypertable *ht, int dimension_index,
 	return max_value;
 }
 
+/*
+ * Get the min value of an open dimension for the hypertable based on the dimension slice info
+ * Note: only takes non-tiered chunks into account.
+ */
+int64
+ts_hypertable_get_open_dim_min_value(const Hypertable *ht, int dimension_index, bool *isnull)
+{
+	StringInfo command;
+	const Dimension *dim;
+	int res;
+	bool min_isnull;
+	Datum mindat;
+	Oid timetype;
+
+	const char *query_str = " SELECT min(dimsl.range_start) FROM _timescaledb_catalog.chunk srcch \
+                   INNER JOIN _timescaledb_catalog.hypertable ht ON ht.id = srcch.hypertable_id  \
+      INNER JOIN _timescaledb_catalog.chunk_constraint chcons ON srcch.id = chcons.chunk_id \
+      INNER JOIN _timescaledb_catalog.dimension dim ON srcch.hypertable_id = dim.hypertable_id \
+      INNER JOIN _timescaledb_catalog.dimension_slice dimsl ON dim.id = dimsl.dimension_id \
+        AND chcons.dimension_slice_id = dimsl.id \
+        AND dimsl.id = %d and ht.id = %d  AND srcch.osm_chunk = false";
+
+	dim = hyperspace_get_open_dimension(ht->space, dimension_index);
+
+	if (NULL == dim)
+		elog(ERROR, "invalid open dimension index %d", dimension_index);
+
+	timetype = ts_dimension_get_partition_type(dim);
+
+	/*
+	 * Query for the oldest chunk in the hypertable.
+	 */
+	command = makeStringInfo();
+	appendStringInfo(command, query_str, ht->fd.id, dim->fd.id);
```

Comment:
Is this correct arguments order? From looking at the query string it seems `dim->fd.id` should come first.

Reply:
nit: `psprintf` would do the same as `StringInfo` here, but would be shorter.